src/chrony.c

```diff
@@ -0,0 +1,1147 @@
+/* chrony plugin for collectd (monitoring of chrony time server daemon)
+ **********************************************************************
+ * Copyright (C) Claudius M Zingerli, ZSeng, 2015-2016 
+ *
+ * Internas roughly based on the ntpd plugin
+ * Some functions copied from chronyd/web (as marked)
+ * 
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of version 2 of the GNU General Public License as
+ * published by the Free Software Foundation.
+ * 
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ * 
+ **********************************************************************
+*/
+/* Formated using 'GNU indent -bli0 -ce -ppi 2' and some manual alignment
+ * TODO:
+ *	- More robust udp parsing (using offsets instead of structs?)
+ *	  -> Currently chrony parses its data the same way as we do (using structs)
+ *	- Plausibility checks on values received
+ *	  -> Done at higher levels
+ */
+
+#include "config.h"
+
+#if HAVE_SYS_TYPES_H
```

Comment:
Sorry, I don't understand what you mean. HAVE_SYS_TYPES_H is usually defined in collectd.h, but there are no includes in config.h.